### PR TITLE
Allow LineAnimator to accept a varying x-axis

### PR DESCRIPTION
**Pull request by DanRyanIrish**
This PR aims to enable the LineAnimator object to handle an x-axis that varies as the animation moves through the slider.

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
This first working version allows the ```axes_ranges``` kwarg to accept arrays of the same shape as the data, where the ```plot_axis_index```th element of the array represents the x-axis, while other dimensions represent duplications of the x-axis values including any changes.

Also note, that in this version ```LineAnimator.update_plot``` explicitly assumes the data is 2D and the last axis is being plotted.  This must be fixed.

**Comment by DanRyanIrish** [comment]
This error doesn't seem to be do with this PR?
https://travis-ci.org/sunpy/sunpy/jobs/347929335

File: examples/LineAnimator_examples.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+This example shows off some ways in which you can use the
+LineAnimator object to animate line plots.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sunpy.visualization.imageanimator import LineAnimator
+
+# Example 1: Animate a 2D cube of random data as a line plot along an
```
Comment (Cadair) [inline examples/LineAnimator_examples.py]: https://github.com/sunpy/sunpy/blob/master/examples/great_arc_example.py#L21

Comment (Cadair) [reply examples/LineAnimator_examples.py]: http://docs.sunpy.org/en/stable/generated/gallery/great_arc_example.html#sphx-glr-generated-gallery-great-arc-example-py

File: sunpy/visualization/imageanimator.py [hunk 2]
```diff
@@ -824,29 +824,31 @@ class LineAnimator(ArrayAnimator):
     def __init__(self, data, plot_axis_index=-1, axis_ranges=None, ylabel=None, xlabel=None,
                  xlim=None, ylim=None, **kwargs):
         # Check inputs.
-        plot_axis_index = int(plot_axis_index)
-        if plot_axis_index not in range(-2, 2):
+        self.plot_axis_index = int(plot_axis_index)
+        if self.plot_axis_index not in range(-2, 2):
```
Comment (Cadair) [inline sunpy/visualization/imageanimator.py]: why is this `(-2, 2)`?

File: examples/LineAnimator_examples.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+This example shows off some ways in which you can use the
+LineAnimator object to animate line plots.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sunpy.visualization.imageanimator import LineAnimator
+
+###############################################################################
+# Example 1: Animate a 2D cube of random data as a line plot along an
```
Comment (Cadair) [inline examples/LineAnimator_examples.py]: I would remove "Example 1:" here.

**Pull request merged**